### ISSUE TITLE
How to start a initial test with a small batch

### ISSUE THREAD

**Issue by 123wula** [ultralytics/ultralytics#21567]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I want to test whether my data and model are ready for training, but I'm unsure how to implement this test. Suggestions include using 8–16 images covering all classes and disabling all data augmentation. However, the results were surprisingly poor, and I'm unclear how to practically apply this advice.​​

​​Here is my command line:​
`cmd
yolo detect train data=data.yaml  cfg=hyp_custom.yaml
`

Here is my hyp_custom.yaml
`model: yolov10n.yaml  # 模型结构
pretrained: null      # 禁用预训练权重（测试学习能力）
freeze: []            # 不解冻任何层

epochs: 100          # 延长训练轮次
batch: 1            # 小批量大小
imgsz: 640           # 图像尺寸
device: 0            # GPU设备

lr0: 0.1             # 初始学习率（提高10倍）
lrf: 1.0             # 最终学习率倍数（保持恒定）
momentum: 0.9        # SGD动量
weight_decay: 0.0    # 禁用权重衰减

optimizer: SGD        # 使用SGD优化器
cls: 1.0             # 分类损失权重
box: 1.0             # 边界框损失权重
dfl: 1.0             # DFL损失权重

hsv_h: 0.0           # 色调增强
hsv_s: 0.0           # 饱和度增强
hsv_v: 0.0           # 明度增强
translate: 0.0       # 平移增强
scale: 0.0           # 缩放增强
flipud: 0.0          # 上下翻转
fliplr: 0.0          # 左右翻转
mosaic: 0.0          # 马赛克增强
mixup: 0.0           # MixUp增强
degrees: 0.0         # 旋转角度
shear: 0.0           # 剪切变换

verbose: True        # 详细输出
patience: 100        # 早停耐心值
save_period: -1      # 不保存中间模型
plots: True          # 生成图表
exist_ok: True       # 覆盖现有结果
`

Here is my data.yaml
`data.yaml
train: D:\YOLO\dataset_coco8_cropped\train\images
val: D:\YOLO\dataset_coco8_cropped\val\images
test: D:\YOLO\dataset_coco8_cropped\test\images  # 如果有测试集

channels: 1#actual data: 1 channel jpgs
nc: 5  # 类别数量（根据你的实际类别数）
names: ['person', 'cyclist', 'car', 'truck', 'bus']  # 38个类别名称的列表
`
Here is part of result(no change😢😔)
train_loss0.66944；train/cls_loss=32.0376；metrics/mAP50(B)=0

**Comment by UltralyticsAssistant** [comment]
👋 Hello @123wula, thank you for reaching out and providing detailed information! 🚀 This is an automated response to help get you started—an Ultralytics engineer will review your issue and assist you soon.

We recommend visiting the [Docs](https://docs.ultralytics.com/) for guidance on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, as well as best practices for testing and training. Many common questions are addressed there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that clearly demonstrates the issue to help us debug efficiently.

If you’re seeking advice on custom training or data testing, please continue to include as much information as possible—such as dataset image examples, training logs, and confirmation you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community in the way that works best for you! For real-time chat, visit [Discord](https://discord.com/invite/ultralytics) 🎧, for in-depth discussions check [Discourse](https://community.ultralytics.com/), or explore threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to connect with others.

## Upgrade

Please upgrade to the latest `ultralytics` package (including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml)) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure your issue isn’t already resolved:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date verified environments, with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If the badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your detailed report—your input helps us improve! An Ultralytics engineer will follow up with you soon.

**Comment by 123wula** [comment]
[CS231n 用于计算机视觉的深度学习](https://cs231n.github.io/neural-networks-3/#baby)
and
"overfit one batch. Overfit a single batch of only a few examples (e.g. as little as two). To do so we increase the capacity of our model (e.g. add layers or filters) and verify that we can reach the lowest achievable loss (e.g. zero). I also like to visualize in the same plot both the label and the prediction and ensure that they end up aligning perfectly once we reach the minimum loss. If they do not, there is a bug somewhere and we cannot continue to the next stage."[from [https://karpathy.github.io/2019/04/25/recipe/](url)]
This is what I want to practice actually.

**Comment by Y-T-G** [comment]
Ultralytics uses TAL which has dynamic assignments based on predictions. I think due to that, it can't be overfit to a small dataset. You can try with `nbs=1` and `mosaic=0`.

**Comment by 123wula** [comment]
@Y-T-G 

Thanks a lot for your help! :blush:  
I practiced your tips and obtained a series of results. To isolate dataset variables, I first trained using COCO8 (8 images).  

Settings:  
data.yaml → yolov10n.yaml (complete)  
```
# hyp.yaml
model: yolov10n.yaml
data: yolov10n.yaml
pretrained: null
freeze: []

task: detect
mode: train

epochs: 200
batch: 1
imgsz: 640
device: 0

name: lr001

lr0: 0.001
lrf: 1.0
momentum: 0.9
weight_decay: 0.0
nbs: 1

optimizer: Adam
box: 7.5
cls: 0.5
dfl: 1.5

erasing: 0.0
auto_augment: none
warmup_epochs: 0.0

hsv_h: 0.0
hsv_s: 0.0
hsv_v: 0.0
translate: 0.0
scale: 0.0
flipud: 0.0
fliplr: 0.0
mosaic: 0.0
mixup: 0.0
degrees: 0.0
shear: 0.0

verbose: True
patience: 100
save_period: -1
plots: True
exist_ok: True
```

The results were somewhat suboptimal:  

1. Briefly: Performance started poorly, and by epoch 200 the metrics fell somewhat short of expectations.
for short it end up like:(The last of 200 epochs)
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=zh-CN style='font-family:Calibri;font-size:11.0pt'>


<p style='margin:0in;font-family:Calibri;font-size:11.0pt'>&nbsp;</p>

<div style='direction:ltr'>


train/box_loss | train/cls_loss | train/dfl_loss | metrics/precision(B) | metrics/recall(B) | metrics/mAP50(B) | metrics/mAP50-95(B)
-- | -- | -- | -- | -- | -- | --
0.25345 | 0.16923 | 1.5823 | 0.42222 | 0.425 | 0.45706 | 0.41538
val/box_loss | val/cls_loss | val/dfl_loss | lr/pg0 | lr/pg1 | lr/pg2 |  
2.0405 | 6.1367 | 3.25152 | 0.001 | 0.001 | 0.001 |  



</div>


</body>

</html>

2. Ideal(maybe) behavior:(Got from Deepseek)
Training Box Loss​​: Must decrease continuously, ultimately approaching zero.

​​Training Cls Loss​​: Must decrease continuously, ultimately approaching zero (for small datasets with clear positive/negative samples).

​​Training DFL Loss​​: Must decrease continuously.

​​Training mAP@0.5(or mAP@0.5:0.95)​​: Core metric! Must increase continuously, ultimately exceeding 0.95 (ideally 0.99+). If mAP gets stuck at a low value (e.g., <0.8) and fails to rise further, it indicates a serious issue with the model.
3. However, my dataset’s performance aligns with the benchmarks mentioned above, ​​except for higher train/box_loss and train/cls_loss​​ — though this observation may not be particularly meaningful.
4. Maybe I can start train?I'm really hoping I don't crash and burn or do all this for nothing.
5. Again,thank you very much! 😊
